sharding-core/sharding-core-merge/src/main/java/org/apache/shardingsphere/core/merge/dal/show/ShowDatabasesMergedResult.java

```diff
@@ -46,6 +49,25 @@
     public ShowDatabasesMergedResult() {
         this(Collections.singletonList(ShardingConstant.LOGIC_SCHEMA_NAME));
     }
+
+    public ShowDatabasesMergedResult(final List<String> schemas) {
+        this.schemas = schemas;
+        this.currentIndex = 0;
+    }
+
+    public ShowDatabasesMergedResult(final ShardingRule shardingRule, final List<QueryResult> queryResults)throws SQLException {
+        this(convertToScheme(queryResults));
+    }
+
+    private static List<String> convertToScheme(final List<QueryResult> queryResults) throws SQLException {
+        final LinkedList<String> result = new LinkedList<>();
+        for (QueryResult queryResult : queryResults) {
+            while (queryResult.next()) {
+                result.add((String) queryResult.getValue(1, String.class));
+            }
+        }
+        return new ArrayList<>(result);
```

Comment:
Why new ArrayList here?
Is it ok if use `new ArrayList<>(queryResults.size())` instead of `new LinkedList<>()` at line 63?

Reply:
queryResults size() is merged two ResultSet... this is size should be the Sum of  row of ResultSet Size , but ResultSet not size function, it's stream reader... so ....